modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -706,7 +711,37 @@ else if (type.equals(DDMFormFieldTypeConstants.SELECT)) {
 						_jsonFactory.createJSONArray(valueString)));
 			}
 			else {
-				if (type.equals(DDMFormFieldTypeConstants.RICH_TEXT)) {
+				if (type.equals(DDMFormFieldTypeConstants.DATE)) {
+					try {
+						Date date = null;
+
+						if (!Validator.isBlank(valueString)) {
+							date = DateUtil.parseDate(
+								"yyyy-MM-dd", valueString, locale);
+						}
+
+						document.addDate(name.concat("_date"), date);
+					}
+					catch (ParseException parseException) {
+						throw new PortalException(parseException);
+					}
+				}
+				else if (type.equals(DDMFormFieldTypeConstants.DATE_TIME)) {
+					try {
+						Date date = null;
+
+						if (!Validator.isBlank(valueString)) {
+							date = DateUtil.parseDate(
+								"yyyy-MM-dd hh:mm", valueString, locale);
+						}
+
+						document.addDate(name.concat("_date"), date);
+					}
+					catch (ParseException parseException) {
+						throw new PortalException(parseException);
+					}
+				}
```

Comment:
@jorgediaz-lr can we reuse code here?

```
if (type.equals(DDMFormFieldTypeConstants.DATE) ||
	type.equals(DDMFormFieldTypeConstants.DATE_TIME)) {

	try {
		String pattern = "yyyy-MM-dd";

		if (type.equals(DDMFormFieldTypeConstants.DATE_TIME)) {
			pattern = "yyyy-MM-dd hh:mm";
		}

		if (!Validator.isBlank(valueString)) {
			document.addDate(
				name.concat("_date"),
				DateUtil.parseDate(
					pattern, valueString, locale));
		}
	}
	catch (ParseException parseException) {
		throw new PortalException(parseException);
	}
}
```

Reply:
ok, I will apply that change

Reply:
See commit b7bf740450e8f2fb4c25868db65a7c5f4dc3849c